### ISSUE TITLE
Load_document/load_video... nothing happened in lua script

### ISSUE THREAD

**Issue by henryaukc** [vysheng/tg#1321]
i tried to use load_documet(msg) or load_documet(msg.id) but not succeed and no error messages. any advice? Thanks a lot

**Comment by henryaukc** [comment]
if msg.media.type == 'document' then
            load_document(msg.id)
            t='Receiving Document ('..msg.id..')\n"'..'"'
            print(t)
			send_msg(msg.from.print_name, t, ok_cb, false)
       	end

-- The code above show no error messages and did nothing. I can't find the file in ~/.telegram-cli/downloads

**Comment by adoob** [comment]
Hi.
You must set up callback to load_document function.
For example:
load_document(msg.id, my_cb, msg)
after loading complete, my_cb will called with args: extra (third arg - msg - from load_document call), bool success (1 or 0) and downloaded filename (if success) as third arg.
this behavior relevant to most tg function calls from lua.

**Comment by b87a94fabea825114134** [comment]
and where is the original name of the file stored to have access to it within the lua script?

**Comment by adoob** [comment]
@b87a94fabea825114134 in third arg of callback func, as said

**Comment by b87a94fabea825114134** [comment]
yes, but the name stored in the last parameter is the name of the actual file that has been stored in the downloads folder. However, the original name of the file where is stored?
for example: I send "thing.pdf" and the 3rd parameter gives me ".telegram-cli/downloads/download_6021512911694857142.pdf"
where is the original name "thing" ?

**Comment by adoob** [comment]
"file_name" field of msg entity - only for "document" type.
video/audio/photo does not have this field.

**Comment by sgregori** [comment]
No file name value found on document type. Any help?

**Comment by joshwapohlmann** [comment]
No, this is not possible with the script. If you have a look at interface.c:3842 you'll see that the document details are hidden in a `tgl_message_media` struct. The original file name can be found in ->document->caption.

this information is NOT included or forwarded, neither in `on_msg_receive` nor the callback. :(
